src/query/index_task.rs

```diff
@@ -0,0 +1,111 @@
+use crate::cluster::{Cluster};
+use std::sync::Arc;
+use crate::errors::{ErrorKind, Result};
+use crate::{ResultCode};
+
+
+
+
+/// Instantiate an IndexTask instance to query Aerospike database cluster for index creation
+/// status.
+#[derive(Debug, Clone)]
+pub struct IndexTask {
+    cluster: Arc<Cluster>,
+    namespace: String,
+    index_name: String
+}
+
+#[derive(Debug, Clone, Copy)]
+pub enum IndexTaskResult {
+    NotFound,
+    InProgress,
+    Complete
+}
+
+static SUCCESS_PATTERN: &'static str = "load_pct=";
+static FAIL_PATTERN_201: &'static str = "FAIL:201";
+static FAIL_PATTERN_203: &'static str = "FAIL:203";
+static DELMITER: &'static str = ";";
+
+impl IndexTask {
+
+
+    // TODO: enforce access of this only to Client
+    /// Initializes IndexTask from client, creation should only be expose to Client
+    pub fn new(cluster: Arc<Cluster>, namespace: String, index_name: String) -> Result<Self> {
+        Ok(IndexTask {
+        	cluster: cluster,
+        	namespace: namespace,
+            index_name: index_name
+        })
+    }
+
+
+    /// Query the status of index creation across all nodes
+    pub fn query_status(&self) -> Result<IndexTaskResult> {
+    	let nodes = self.cluster.nodes();
+
+    	if nodes.len() == 0 {
+    		// TODO: what error should this be
+    	    bail!(ErrorKind::ServerError(ResultCode::ServerError))
+    	}
```

Comment:
I'm not convinced that this should be an error at all! Definitely not a server error, which should only be returned if the client received an error response from the server with a specific error code.

How would this condition occur? Maybe if you issue the create index command, then close the client, and then try to query the index task? If that's the case, maybe we should introduce a new "connection closed" error, indicating that the cluster object referenced by the task is no longer connected to the server?

Reply:
Hmm, its definitely possible to reach this branch, so I favor having a define behavior, I like your idea of connection closed error let me try that out.